modules/API/external/external_api.js

```diff
@@ -259,6 +266,11 @@ export default class JitsiMeetExternalAPI extends EventEmitter {
         this._frame.setAttribute('allowFullScreen', 'true');
         this._frame.style.border = 0;
         this._frame = this._parentNode.appendChild(this._frame);
+
+        // waits for iframe resources to load and fires event when it is done
+        this._frame.contentWindow.addEventListener('load', () => {
+            this.emit(UI_LOADED_EVENT);
+        });
```

Comment:
Adding the listener here changes the pattern of the event emitting a bit. Before all events were being fired by jitsi-meet somehow and then piped through external_api. This change seems okay to me really but I'd like to your know your thoughts on this inconsistency.

Reply:
Yep, this is intentional. If there is some external UI showing the loading of the page, it will listen for this event to close itself.
And in the case where for example app.bundle is not loaded, there is no way to fire that event, but it will anyway load something(the 'fatal' page saying that this resource was not loaded) which will instruct the UI using the external api to close whatever was shown and show the UI coming from meet.